From cc27fcba9e5e2d7b7cd62e91d67b96cc47ee32b2 Mon Sep 17 00:00:00 2001
From: Geetha sowjanya <gakula@marvell.com>
Date: Fri, 26 Jul 2019 17:27:56 +0530
Subject: [PATCH 366/386] PCI: quirks : Apply ACS quirk for all devices

Due to the current quirk ACS capability is set to only for
the root devices.
Multifunction devices get the same iommu group number if ACS
capability is not set. Hence, changing quirk to set ACS capability
for all devices.

Change-Id: Idff6c92229e5e568db5bc19efce68f983d3d59dc
Signed-off-by: Geetha sowjanya <gakula@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/13115
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Radha Chintakuntla <radhac@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 drivers/pci/quirks.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/drivers/pci/quirks.c b/drivers/pci/quirks.c
index c4c2ce0bae45..9ad2adbafcb6 100644
--- a/drivers/pci/quirks.c
+++ b/drivers/pci/quirks.c
@@ -4316,9 +4316,7 @@ static bool pci_quirk_cavium_acs_match(struct pci_dev *dev)
 	 * bits of device ID are used to indicate which subdevice is used
 	 * within the SoC.
 	 */
-	return (pci_is_pcie(dev) &&
-		(pci_pcie_type(dev) == PCI_EXP_TYPE_ROOT_PORT) &&
-		((dev->device & 0xf800) == 0xa000));
+	return (pci_is_pcie(dev) && ((dev->device & 0xf800) == 0xa000));
 }
 
 static int pci_quirk_cavium_acs(struct pci_dev *dev, u16 acs_flags)
-- 
2.17.1

